### PR TITLE
added autoscaling support to Python APIs

### PR DESCRIPTION
**Pull request by blublinsky**
@tedhtchang, @smit-kiri, @anyscalesam

## Why are these changes needed?

This cleans up Python implementation add support for creation of autoscaling clusters and image pull policy

## Related issue number

https://github.com/ray-project/kuberay/issues/2029

## Checks

- [ x] I've made sure the tests are passing.
- Testing Strategy
   - [x ] Unit tests
   - [x ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by blublinsky** [comment]
@tedhtchang, @smit-kiri, @anyscalesam, please take a look

cc @kevin85421

**Comment by smit-kiri** [comment]
Just a general suggestion on the classes defined here - most of them could be defined as `dataclass` [objects](https://docs.python.org/3/library/dataclasses.html). Makes it easier to read / know types of the fields

File: clients/python-apiserver-client/pyproject.toml
Comment (tedhtchang) [inline clients/python-apiserver-client/pyproject.toml:1]: Should we add kubernetes package to the dependencies section

**Review by tedhtchang — APPROVED** [review]
/LGTM I verified I can create RayCluster the AutoscalerOptions and trigger autoscaling worker pod by submitting a job.

File: clients/python-apiserver-client/pyproject.toml
Comment (blublinsky) [reply clients/python-apiserver-client/pyproject.toml:1]: Not really

**Comment by blublinsky** [comment]
@kevin85421 all yours

**Review by kevin85421 — APPROVED** [review]
I don't review the PR, but the PR doesn't touch KubeRay itself. Merge.

**Pull request merged**